pkgs/applications/editors/emacs/generic.nix

```diff
@@ -67,18 +67,18 @@ in stdenv.mkDerivation {
 
     # Make native compilation work both inside and outside of nix build
     (lib.optionalString nativeComp (let
-      libPath = lib.concatStringsSep ":" [
-        "${lib.getLib libgccjit}/lib/gcc/${targetPlatform.config}/${libgccjit.version}"
-        "${lib.getLib stdenv.cc.cc}/lib"
-        "${lib.getLib stdenv.glibc}/lib"
-      ];
+      libPath = (lib.concatStringsSep " "
+        (builtins.map (x: ''\"-L${x}\"'') [
```

Comment:
Wondering if we should use -B here

Reply:
Do you mean for binutils's `ld`? Darwin's `ld` does not support any `-B` options, as far as I can tell.

Reply:
I think `-B` is a CC flag, so it's gcc that handles it. I think gccemacs only cares about library files, so I'm not sure if the `-L` vs. `-B` is relevant though.

Reply:
Ah, it may depend on whether libgccjit recognizes that option as a "driver" option (vs. a "frontend" option I guess?) - the [manual warns](https://gcc.gnu.org/onlinedocs/jit/topics/contexts.html#c.gcc_jit_context_add_driver_option) that only options for the assembler or linker will be meaningful as "driver options". Definitely something to experiment with, though!

Reply:
Yes, the guix packaging for native-comp explicitly uses the -B option https://github.com/flatwhatson/guix-channel/commit/f9bf0a476cb37a3718dfb9d6a78f22c33504ae97 else it complains about not finding crti.o, crtbeginS.o, and -lgcc. I haven't been able to compile this succesfully else I'd test.

Reply:
Fascinating! I'll try that - might also help us get rid of the need to use wrapProgram to set paths to the linker and assembler.

Reply:
Ok, just confirmed that on linux building with the -B option is necessary for straight.el to compile packages outside of nix